### PR TITLE
Checkout: remove dependency on lib/cart-values product creators

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This is a follow up to https://github.com/Automattic/wp-calypso/pull/46532 which moved the shopping cart manager to the `@automattic/shopping-cart` package.

In this PR we remove the dependency on the product creation functions from `lib/cart-values` from new checkout, which should hopefully allow us to eventually remove many parts of that directory and rely instead on the `@automattic/shopping-cart` package. 

Previously, checkout would use functions like `planItem()` and `domainItem()` from `lib/cart-values` to format data (mostly the product slug, but also other associated data like a domain name) into a `RequestCartProduct` prior to adding those products to the cart. Most of these functions, however, are simple mappings of the form:

```js
(product_slug, product_id, meta) => ({product_slug, product_id, meta})
```

In `usePrepareProductsForCart` (the primary consumer of these functions in composite checkout) we already have the product slug and ID and knowing what kind of product we're creating is not as important except for products which require special properties.

In this PR we add a single helper function, `createRequestCartProduct()`, which makes sure that all the required fields are set for a `RequestCartProduct` and fills in the defaults. This proved enough to replace all uses of the previously mentioned functions within the `my-sites/checkout/composite-checkout` directory.

This does not change any uses of the `lib/cart-values` functions from other parts of calypso, including those that might be used by components imported into new checkout. That will have to come later.

#### Testing instructions

This modifies how products are added to the cart from the URL when checkout loads. Therefore it's worth testing all the ways that this happens and verifying that the correct products are added each time.

- Use a URL which adds a plan to the cart via its alias like `/checkout/example.com/premium`.
- Use a URL which adds a non-plan product to the cart like `/checkout/example.com/domain-mapping:example.com`.
- Use a URL which adds a renewal product to the cart like `/checkout/value_bundle/renew/1234567/example.com`.

### CODE REVIEW

**Comment by sirbrillig** [comment]
"Cannot find .sidebar" e2e test failures are expected and should be fixed by https://github.com/Automattic/wp-calypso/pull/46711

**Comment by nbloomf** [comment]
Here are the results of my testing. TL;DR: it works as expected.

- *Add plan item via the URL using an alias*:
    - Adding a personal plan to a free site loaded checkout as expected.
    - Adding a personal plan to a premium site loaded checkout with the personal plan but prevented purchase at the pay button with a relevant error message. This is "expected", but it would probably be good to detect this situation earlier.
    - Adding a premium plan to a premium site loaded checkout with the premium plan, but did not convert it to a renewal, as the "Renew" label on the line item was not present. I think the backend handles this correctly anyway, but it would be nice to make it clear when purchasing a plan when the site already has that plan that the purchase will be a renewal.
- *Add non-plan item via the URL*:
    - Adding a domain mapping to a free site redirected to the plans page as expected (since free sites cannot have domain mappings without a plan).
    - Adding a domain mapping to a premium site without a domain loaded checkout with the correct cart items and the free purchase method.
    - Adding a second domain mapping to a premium site (that already has one) loaded correctly.
- *Add a renewal via the URL*:
    - Checkout loaded with the renewal item as expected.

File: client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts [hunk 1]
```diff
@@ -407,15 +309,14 @@ function useAddProductFromSlug( {
 			return;
 		}
 
-		const cartProducts = validProducts
-			.map( ( product ) =>
-				createItemToAddToCart( {
-					productAlias: product.product_alias,
-					product_id: product.product_id,
-					isJetpackNotAtomic,
```
Comment (nbloomf) [inline client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts:404]: nice.

**Review by nbloomf — APPROVED** [review]
Looks good and works as described!

**Pull request merged**